hudi-common/src/main/java/org/apache/hudi/common/table/view/AbstractTableFileSystemView.java

```diff
@@ -245,7 +245,7 @@ public final void reset() {
       bootstrapIndex = null;
 
       // Initialize with new Hoodie timeline.
-      init(metaClient, getTimeline());
+      init(metaClient, metaClient.reloadActiveTimeline());
```

Comment:
IIUC, this is breaking some fundamental assumptions.  There are many places where we pass "trimmed" timeline for time-travel queries etc. You are replacing that with all instants from active timeline, which is not desired.

Is this change needed if we handle empty partitionToReplaceFileIds in archival?

Reply:
@satishkotha 
Since this part will called after archival, and the archived commits still in the timeline.
In the post process hudi will try to load the byte from them, and will cause IO error

Reply:
I think the root problem is why we are calling reset() when close the timeline

Reply:
@bvaradar mentioned this is by design. Balaji, could you please help resolve this?

Reply:
I've put `getTimeline()` back since it should be irrelevant to the archival issue though I am still curious to learn why this is needed.